File: 1288-remove-covered-intervals/1288-remove-covered-intervals.py
class Solution:
    def removeCoveredIntervals(self, A: list[list[int]]) -> int:
        remain = len(A)
        A.sort(key=lambda x: (x[0], -x[1]))
        for i in range(len(A)):
            for j in range(len(A)):
                if i == j:
                    continue
                if A[i][0] >= A[j][0] and A[i][1] <= A[j][1]:
                    remain -= 1
                    break
        
        return remain

File: 1288-remove-covered-intervals/1288-remove-covered-intervals.cpp
class Solution {
public:
    int removeCoveredIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), [](vector<int>& a, vector<int>& b){
            if(a[0] == b[0]) return a[1] > b[1];
            return a[0] < b[0];
        });
        vector<vector<int>> ans;
        for(int i = 0; i < intervals.size(); i++){
            if(i > 0 &&
               ans.back()[0] <= intervals[i][0] &&
               ans.back()[1] >= intervals[i][1]){
                continue;
            }
            ans.push_back(intervals[i]);
        }

        return ans.size();
    }
};